Add Agent hardware and concurrency clarifications (#1723)
hashicorp/terraform-website: content/source/docs/cloud/agents/index.html.md

Agents support Terraform versions 0.12 and above. Workspaces configured to use Terraform versions below 0.12 will not be able to select the agent-based execution mode.

### Hardware Requirements

> The host running the agent will have varying resource requirements depending on the workspace being run.  A host for an agent can be a dedicated or shared cloud instance, virtual machine, bare metal server, or a container.  Memory, CPU and disk space should be monitored and adjusted accordingly as usage and performance will vary upon the workspaces being executed. The specs below are provided as a reference. Depending on the type of environment you deploy to, the name of your instance type may vary.

Reviewer: Just thinking about active voice, flow (I want the specs sentence to come right before the specs), and concision. Please make sure I didn't materially change the meaning!

```suggestion
The host running the agent will have varying resource requirements depending on the workspace. A host can be a dedicated or shared cloud instance, virtual machine, bare metal server, or a container. You should monitor and adjust memory, CPU, and disk space based on each workspace's usage and performance. The name of your instance type may vary depending on your deployment environment. 

The specs below are provided as a reference. 
```

Agents support Terraform versions 0.12 and above. Workspaces configured to use Terraform versions below 0.12 will not be able to select the agent-based execution mode.

### Hardware Requirements

> The instance running the agent will consume varying amounts of resources depending on the workspace being run. Memory, CPU and disk space should be monitored on the agent instance and adjusted accordingly. Below is provided as a typical reference instance size.

Reviewer 2: ```suggestion
The instance running the agent will have varying resource requirements depending on the workspace being run. Memory, CPU and disk space should be monitored on the agent instance and adjusted accordingly. The specs below are provided as a reference. Depending on the type of environment you deploy to, the name of your instance may vary.
```

> The instance running the agent will have varying resource requirements depending on the workspace being run. Memory, CPU and disk space should be monitored on the agent instance and adjusted accordingly. The specs below are provided as a reference. Depending on the type of environment you deploy to, the name of your instance may vary.

Reviewer 2: ```suggestion
The instance running the agent will have varying resource requirements depending on the workspace being run. Memory, CPU and disk space should be monitored on the agent instance and adjusted accordingly. The specs below are provided as a reference. Depending on the type of environment you deploy to, the name of your instance type may vary.
```

Reviewer 2: I'm not sure if I missed this word in the last review, but it's important here

* At least 4GB of free disk space

>   * Each run requires the agent to locally store the tarred repository, extracted repository, state file, providers and the terraform binary. For example, a workspace with hundreds of resources and multiple providers can easily consume 500MB of disk space.

Reviewer 2: ```suggestion
  * Each run requires the agent to temporarily store local copies of the tarred repository, extracted repository, state file, any providers or modules, and the Terraform binary itself.
```

* At least 2GB of system memory

> * At least 2 CPU cores

Reviewer 3: Unless we have proof around the number of CPU cores, I would separate it from the min requirements as a note that calls out Terraform will spread out across cores if they are made available.

Author: My logic on that was with 1 core that the agent, terraform main proc, forked procs, and the providers all compete and thrash for CPU and things slowed down a bit when I was playing locally (non cloud stuff).
I think we can remove it - if run on 1 core it works, just slower which should be obvious to most.

You may choose to run multiple agents within your network, up to the organization's purchased agent limit. If there are multiple agents available within an organization, Terraform Cloud will select the first available agent within the target pool.

> Each agent process will run a single Terraform run at a time. Multiple agent processes can be concurrently run on a single instance. 

Reviewer 2: ```suggestion
Each agent process will run a single Terraform run at a time. Multiple agent processes can be concurrently run on a single instance, license limit permitting. 
```
